docs/src/main/asciidoc/security-proactive-authentication.adoc

```diff
@@ -30,28 +30,28 @@ If you disable proactive authentication, the authentication process runs only wh
 An identity can be requested because of security rules that require the user to authenticate or because programmatic access to the current identity is required.
 
 If proactive authentication is in use, accessing `SecurityIdentity` is a blocking operation.
-This is because authentication might have yet to happen and accessing `SecurityIdentity` might require calls to external systems, such as databases, that might block the operation.
+This is because authentication might have yet to happen, and accessing `SecurityIdentity` might require calls to external systems, such as databases, that might block the operation.
 For blocking applications, this is not an issue.
-However, if you have disabled authentication in a reactive application, this will fail because you cannot do blocking operations on the I/O thread.
+However, if you have disabled authentication in a reactive application, this fails because you cannot do blocking operations on the I/O thread.
 To work around this, you need to `@Inject` an instance of `io.quarkus.security.identity.CurrentIdentityAssociation` and call the `Uni<SecurityIdentity> getDeferredIdentity();` method.
-Then, you can subscribe to the resulting `Uni` and will be notified when authentication is complete and the identity is available.
+Then, you can subscribe to the resulting `Uni` to be notified when authentication is complete and the identity is available.
 
 [NOTE]
 ====
 You can still access `SecurityIdentity` synchronously with `public SecurityIdentity getIdentity()` in xref:resteasy-reactive.adoc[RESTEasy Reactive] from endpoints that are annotated with `@RolesAllowed`, `@Authenticated`, or with respective configuration authorization checks because authentication has already happened.
 The same is also valid for xref:reactive-routes.adoc[Reactive routes] if a route response is synchronous.
 ====
 
-xref:security-authorize-web-endpoints-reference.adoc#standard-security-annotations[Standard security annotations] on CDI beans are not supported on an I/O thread if a non-void secured method returns a value synchronously and proactive authentication is disabled because they need to access `SecurityIdentity`.
+xref:security-authorize-web-endpoints-reference.adoc#standard-security-annotations[Standard security annotations] on CDI beans are not supported on an I/O thread if a non-void secured method returns a value synchronously and proactive authentication is disabled because they must access `SecurityIdentity`.
```

Comment:
Does it helpt to rephrase this? 
Suggestion A:
```suggestion
If a non-void secured method returns a value synchronously and proactive authentication is disabled, xref:security-authorize-web-endpoints-reference.adoc#standard-security-annotations[standard security annotations] on CDI beans are not supported on an I/O thread because they must access `SecurityIdentity`.
```
Suggestion B:
```suggestion
When proactive authentication is disabled, xref:security-authorize-web-endpoints-reference.adoc#standard-security-annotations[standard security annotations] used on CDI beans do not function on an I/O thread if a secured method that is not void synchronously returns a value. This limitation arises from the necessity for these methods to access `SecurityIdentity`.
```
What do you think, @sberyozkin ?

Reply:
Hi @rolfedh, suggestion B reads easier to me, thanks

Reply:
Updated. Thanks!